### PR TITLE
DataSpec

### PR DESCRIPTION
**Pull request by orestisfoufris**
Initial commit of unit tests for Data.hs

### CODE REVIEW

File: src/Network/Haskoin/Store/Data.hs [hunk 1]
```diff
@@ -131,15 +131,32 @@ instance JsonSerial TxHash where
...
+        Left _ -> mzero
+        Right v -> return v
+      case stringToAddr net t of
+        Nothing -> mzero
+        Just x -> return x
+
 class BinSerial a where
     binSerial :: Network -> Putter a
 
+class BinDeserial a where
```
Comment (xenog) [inline src/Network/Haskoin/Store/Data.hs]: Merge the `BinDeserial` and `BinSerial` classes.

Comment (orestisfoufris) [reply src/Network/Haskoin/Store/Data.hs]: done!

File: test/Network/Haskoin/Store/DataSpec.hs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  let net = btc
+      def = TxHash "75ad09e40a8246beed18646c393b8687f084a23f1336179349f6891b6028e974"
+      defAddr = fromJust $ stringToAddr net "1Mfn26ABTDBexhSUbsE3MxKBLSHfHQj9Z7"
+
+  describe "Test TxHash serialization" $
+    it "serializes tx" $ do
+      let tx = TxHash "0666939fb16533c8e5ebaf6052bb8c90d27ee53fe6035bb763de5253e0b1cd44"
+          raw = runPut $ binSerial net tx
+          deserial = runGet (binDeserial net) raw
+      fromRight def deserial `shouldBe` tx
```
Comment (xenog) [inline test/Network/Haskoin/Store/DataSpec.hs]: Create a single generic function that takes something with an instance of `BinSerial`, plus a `Network`, and does the test as such:

```haskell
let serial = runPut . binSerial net
    deserial = runGet (binDeserial net)
(deserial . serial) x `shouldBe` Right x
```

File: test/Network/Haskoin/Store/DataSpec.hs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  describe "Test Address serialization" $
+    it "serialize address" $ do
+      let addr = fromJust $ stringToAddr net "1DtDAYYTWRoiXvHRjARwVhjCUnNTk1XfXw"
+          raw = runPut $ binSerial net addr
+          deserial = runGet (binDeserial net) raw
+      fromRight defAddr deserial `shouldBe` addr
```
Comment (xenog) [inline test/Network/Haskoin/Store/DataSpec.hs]: Use that function which includes code roughly following the previous example to then do all your comparisons, such that you can have much shorter specs for every test.

File: test/Network/Haskoin/Store/DataSpec.hs (hunk 2 above)
Comment (xenog) [reply test/Network/Haskoin/Store/DataSpec.hs]: Function signature should look like:

```haskell
testSerial :: BinSerial a => Network -> a -> Bool
```

Comment (xenog) [reply test/Network/Haskoin/Store/DataSpec.hs]: Well, if you use `shouldBe` then the signature will be different:

```haskell
testSerial :: BinSerial a => Network -> a -> Expectation
```

**Pull request closed without merging**